src/ShopBundle/objects/db/TShopOrder.class.php

```diff
@@ -488,6 +488,7 @@ public function MarkOrderAsExportedToWaWi($bExported = true)
      */
     public function SendOrderNotification($sSendToMail = null, $sSendToName = null)
     {
+        $this->SetSendOrderNotificationState(true);
```

Comment:
I don't really see the "semaphore" here? Is this a side-effect? Who uses "system_order_notification_send"?

Reply:
If "true" is meaing that the order confirmation was sent isn't this line then a sort-of lie?

Reply:
> If "true" is meaing that the order confirmation was sent isn't this line then a sort-of lie?
It sets only the notification _state_, so the former description/interpretion of the underlaying field may be changed.
A more proper way would be an extra field to perform the "semaphore" behaviour. So this a question of effort versus effect.

Reply:
Code analysis: Most code locations consider this field before calling SendOrderNotification(). Others rightfully don't do it - for example the backend button "send again".